### ISSUE TITLE
launchOrFocus on Finder doesn't work with multiple screens

### ISSUE THREAD

**Issue by oskarols** [Hammerspoon/hammerspoon#304]
Basic test case: have 2 screens, finder on one screen, and have the other screen focused.

Try running:

```
hs.application.launchOrFocus("Finder")
```

Finder will not be focused.

Behind the scenes it seems Finder is running a non-standard window, it seems to be the case that when you do not have an actual  Finder window on the same screen then focus instead goes to this hidden window.

**Comment by cmsj** [comment]
Oddly, this seems to not be reproducible from the Hammerspoon Console (on 10.10.3). I have a Finder window on my second monitor, under several other windows, the Console open on my primary monitor, with focus, and running hs.application.launchOrFocus("Finder") pops the Finder window to the front of the second monitor and gives it focus.

**Comment by oskarols** [comment]
@cmsj I think I was able to do the same thing, as long as the console was *created* in the same window as a Finder window.

**Comment by cmsj** [comment]
@oskarols good point, I was able to reproduce this using the console if it was created on a different screen to a Finder window.

So it definitely looks to me like this is handing focus to the desktop, which is a "special" Finder window.

The question is, what should Hammerspoon be doing about this? I'd really really like to avoid having lots of whitelisting/blacklisting in the APIs. I think I'd prefer to just add a note to the API docs that launchOrFocus() may behave in unusual ways for apps that have "special" windows.

Thoughts?

**Comment by lowne** [comment]
At least over here, the system shortcut ```cmd-` ``` always cycles through (what I presume to be, since the icons mysteriously vanished) the desktop, so it seems indeed a case of working-as-intended.
Then again I don't know what the point of this behaviour is (so IMHO this might deserve an exceptional 'fix' for launchOrFocus) - I mean, if Finder at least triggered the 'show desktop' action, it might make more sense - but there must probably be some (historical) reason.
